index.js

```diff
@@ -41,7 +47,35 @@ program
   .description('process all the fragment include in the given file')
   .option('-o, --output <path>', 'output file path')
   .action((file, options) => {
-    markbinder.includeFile(path.resolve(process.cwd(), file))
+    const config = {};
+    const siteConfigPath = findUp.sync(SITE_CONFIG_NAME);
+    config.rootPath = path.dirname(siteConfigPath);
+    config.baseUrlMap = {};
+    config.baseUrlMap[config.rootPath] = true;
+    config.userDefinedVariablesMap = {};
+
+    const userDefinedVariables = {};
+    let content;
+    try {
+      const userDefinedVariablesPath = path.resolve(config.rootPath, USER_VARIABLES_PATH);
+      content = fs.readFileSync(userDefinedVariablesPath, 'utf8');
+    } catch (e) {
+      content = '';
+      logger.warn(e.message);
+    }
+    const $ = cheerio.load(content);
+    $.root().children().each(function () {
+      const id = $(this).attr('id');
+      const html = $(this).html();
+      userDefinedVariables[id] = html;
+    });
+
+    // This is to prevent the first nunjuck call from converting {{baseUrl}} to an empty string
+    // and let the baseUrl value be injected later.
+    userDefinedVariables.baseUrl = '{{baseUrl}}';
+    config.userDefinedVariablesMap[config.rootPath] = userDefinedVariables;
```

Comment:
This code seems to be a rewrite of `Site#collectUserDefinedVariablesMap` but for a single site. Would it be possible to extract the parsing of `userDefinedVariables`? Possibly in `util` or in `Site` as a static function? That way we also don't have to import cheerio here as well. Additionally, we can easily write tests for the function.